automation/constructTree.sh

```diff
@@ -85,235 +89,247 @@ ACCOUNT_CONFIG_REFERENCE="${ACCOUNT_CONFIG_REFERENCE:-$ACCOUNT_CONFIG_REFERENCE_
+                        ( -d "${ACCOUNT_CANDIDATE_DIR}/config/account.json" )
+                    ) ]]; then
+                    # Multi-account repo
+                    ACCOUNT_CONFIG_DIR="${BASE_DIR}/config/accounts"
+                    MULTI_ACCOUNT_REPO=true
+                else
+                    # Single account repo
+                    ACCOUNT_CONFIG_DIR="${BASE_DIR}/config/${ACCOUNT}"
+                fi
+            fi
+            mkdir -p $(filePath "${ACCOUNT_CONFIG_DIR}")
+            mv "${BASE_DIR_TEMP}" "${ACCOUNT_CONFIG_DIR}"
+            save_context_property ACCOUNT_CONFIG_COMMIT "$(git -C "${ACCOUNT_CONFIG_DIR}" rev-parse HEAD)"
+        fi
+
+        ACCOUNT_INFRASTRUCTURE_DIR=$(findGen3AccountInfrastructureDir "${BASE_DIR}" "${ACCOUNT}")
+        if [[ -z "${ACCOUNT_INFRASTRUCTURE_DIR}" ]]; then
+            # Pull in the account infrastructure repo
+            ${AUTOMATION_DIR}/manageRepo.sh -c -l "account infrastructure" \
+                -n "${ACCOUNT_INFRASTRUCTURE_REPO}" -v "${ACCOUNT_GIT_PROVIDER}" \
+                -d "${BASE_DIR_TEMP}" -b "${ACCOUNT_INFRASTRUCTURE_REFERENCE}"
+            RESULT=$? && [[ ${RESULT} -ne 0 ]] && exit
 
+            # Ensure temporary files are ignored
+            [[ (! -f "${BASE_DIR_TEMP}/.gitignore") || ($(grep -q "temp_\*" "${BASE_DIR_TEMP}/.gitignore") -ne 0) ]] && \
+            echo "temp_*" >> "${BASE_DIR_TEMP}/.gitignore"
+
+            # Is account repo contains multiple accounts, assume the infrastructure repo does too
+            if [[ "${MULTI_ACCOUNT_REPO}" == "true" ]]; then
+                # Multi-account repo
+                ACCOUNT_INFRASTRUCTURE_DIR="${BASE_DIR}/infrastructure/accounts"
+            else
+                # Single account repo
+                ACCOUNT_INFRASTRUCTURE_DIR="${BASE_DIR}/infrastructure/${ACCOUNT}"
+            fi
+            mkdir -p $(filePath "${ACCOUNT_INFRASTRUCTURE_DIR}")
+            mv "${BASE_DIR_TEMP}" "${ACCOUNT_INFRASTRUCTURE_DIR}"
+        fi
+
+    # TODO(mfl): 03/02/2020 Remove the following code once its confirmed its redundant
+    # From the Jenkins logs it throws errors when TENANT is non-empty which makes sense
+    # as BASE_DIR_TEMP has been cleared by processing of the ACCOUNT. However sometimes
+    # it doesn't which suggests that it either is blank or contains a directory with an
+    # infrastructure subdirectory.
+    # Either way, without a temp dir to move, it seems unnecessary.
+    #    TENANT_INFRASTRUCTURE_DIR=$(findGen3TenantInfrastructureDir "${BASE_DIR}" "${TENANT}")
+    #    if [[ -z "${TENANT_INFRASTRUCTURE_DIR}" ]]; then
+    #
+    #        TENANT_INFRASTRUCTURE_DIR="${BASE_DIR}/${TENANT}"
+    #        mkdir -p $(filePath "${TENANT_INFRASTRUCTURE_DIR}")
+    #        mv "${BASE_DIR_TEMP}" "${TENANT_INFRASTRUCTURE_DIR}"
+    #    fi
+
+    fi
+fi
+
+if [[ "${USE_EXISTING_TREE}" == "true" ]]; then
+    if [[ -n "${ROOT_DIR}" && -d "${ROOT_DIR}" ]]; then
```

Comment:
This logic seems back the front? It is reporting an error is the variable is defined and is a directory...

Reply:
```suggestion
    if [[ -z "${ROOT_DIR}" || -d "${ROOT_DIR}" ]]; then
```

Reply:
Yeah sorry that should have been the opposite. Suggested change above If you can check it @ml019